### ISSUE TITLE
fetchMore errors should show up in the QueryResult

### ISSUE THREAD

**Issue by vishwam** [apollographql/apollo-client#6112]
**Intended outcome:**
```ts
const { loading, error, data, fetchMore } = useQuery(...);
return <Feed
        entries={data.feed || []}
        onLoadMore={... // calls fetchMore as described in the documentation
```

When fetchMore is triggered, it sets `loading: true` in the query result (as expected). If the new request fails, the `error` field should be populated.

**Actual outcome:**
fetchMore fails silently, the `error` field is never set.

**How to reproduce the issue:**
I've put a trivial example here: https://codesandbox.io/s/little-meadow-6v505. This deliberately triggers an error in updateQuery, and the expectation is that the error should be propagated to the QueryResult so the component can display it correctly.

This scenario also repros when the network call is actually made and fails (e.g., if the user is offline)

**Versions**
Apollo-Client:  3.0.0-beta.42
System:
    OS: Windows 10 10.0.18363
  Binaries:
    Node: 12.13.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.21.1 - C:\Program Files (x86)\Yarn\bin\yarn.CMD
    npm: 6.9.0 - C:\Program Files\nodejs\npm.CMD
  Browsers:
    Edge: 44.18362.449.0

**Comment by benjamn** [comment]
The error behavior of `fetchMore` changed recently in #6551 with commit 6607e763533a9393c60eafca57169d48cdf97c46. Using the latest RC version, the error behavior in your reproduction seems different, though I'm not sure if it's what you expected. Can you try `@apollo/client@3.0.0-rc.12`?

We're trying to clean up the Release 3.0 milestone in preparation for the launch, so I'm going to close this issue for now, but please feel free to reopen it if you're still seeing this problem.

**Comment by PatoBeltran** [comment]
@benjamn I don't think this was solved with v3.0.0, if you take the provided codesandbox and update to v3.1.4 the error now is just thrown during fetchMore instead of updating queryResults from the original query... although this behavior is better than what it was before (the fetch more can now be wrapped in a try catch and handle the error separately, instead of just silently failing) it still not being correlated back to the original QueryResult (the same way the loading does)
